### PR TITLE
Add support for reusable WebDriver sessions.

### PR DESCRIPTION
**Pull request by DrMarcII**

### CODE REVIEW

File: go/launcher/proxy/driverhub/driver_hub.go [hunk 1]
```diff
@@ -124,12 +126,42 @@ func (h *WebDriverHub) GetActiveSessions() []string {
 	return result
 }
 
-// Shutdown is a no-op on WebDriverHub.
+// Shutdown is shutdowns any running sessions.
```
Comment (juangj) [inline go/launcher/proxy/driverhub/driver_hub.go]: s/is shutdowns/shuts down/

File: go/launcher/proxy/driverhub/driver_hub.go [hunk 2]
```diff
@@ -124,12 +126,42 @@ func (h *WebDriverHub) GetActiveSessions() []string {
...
+			h.reusableSessions = append(h.reusableSessions[:i], h.reusableSessions[i+1:]...)
+			return session, true
+		}
+	}
+	return nil, false
+}
+
+func (h *WebDriverHub) AddReusableSession(session *WebDriverSession) {
+	if !capabilities.CanReuseSession(session.Desired) {
+		return
```
Comment (juangj) [inline go/launcher/proxy/driverhub/driver_hub.go]: Seems like it should be an error or something if you attempt to add a non-reusable session here. I suppose you currently would never use the return value, though.

File: go/launcher/proxy/driverhub/driver_hub.go (hunk 1 above)
Comment (DrMarcII) [inline go/launcher/proxy/driverhub/driver_hub.go]: Done.

File: go/launcher/proxy/driverhub/driver_hub.go (hunk 2 above)
Comment (DrMarcII) [inline go/launcher/proxy/driverhub/driver_hub.go]: Done.

**Comment by DrMarcII** [comment]
jenkins retest this please

**Pull request merged**